### PR TITLE
Add Privilege for Updating Maintenance Properties in Catalog

### PR DESCRIPTION
**Pull request by flyrain**
# Description

This is the first PRs to add table maintenance properties in Polaris. This PRs focus on adding a privilege for catalog write maintenance. Please see more details in this design doc, https://docs.google.com/document/d/1Pd_mzZcfvnUvcH98IbwsIYf4eryet1lQDfclKYx-t-M/edit?usp=sharing

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Added unit test

# Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] If adding new functionality, I have discussed my implementation with the community using the linked GitHub issue

### CODE REVIEW

File: polaris-core/src/main/java/org/apache/polaris/core/entity/PolarisPrivilege.java [hunk 1]
```diff
@@ -102,7 +102,9 @@ public enum PolarisPrivilege {
   CATALOG_ROLE_FULL_METADATA(67, PolarisEntityType.CATALOG_ROLE),
   CATALOG_ROLE_MANAGE_GRANTS_ON_SECURABLE(68, PolarisEntityType.CATALOG_ROLE),
   CATALOG_ROLE_MANAGE_GRANTS_FOR_GRANTEE(69, PolarisEntityType.CATALOG_ROLE),
-  ;
```
Comment (dimas-b) [inline polaris-core/src/main/java/org/apache/polaris/core/entity/PolarisPrivilege.java:105]: nit: keeping this semicolon on a new line allows for pure "add" diffs in PRs and simplifies linking enum values to commits that truly added them :)

File: polaris-core/src/main/java/org/apache/polaris/core/auth/PolarisAuthorizableOperation.java [hunk 2]
```diff
@@ -121,6 +122,7 @@ public enum PolarisAuthorizableOperation {
   CREATE_CATALOG(CATALOG_CREATE),
   GET_CATALOG(CATALOG_READ_PROPERTIES),
   UPDATE_CATALOG(CATALOG_WRITE_PROPERTIES),
+  UPDATE_CATALOG_MAINTENANCE_PROPERTIES(CATALOG_WRITE_MAINTENANCE_PROPERTIES),
```
Comment (dimas-b) [inline polaris-core/src/main/java/org/apache/polaris/core/auth/PolarisAuthorizableOperation.java:125]: tangential: This change LGTM, but I'm thinking about braking down the operation enum into several enums with dedicated scope (operations within a catalog, catalog management (add/remote), ermission management)... Just FYI :)

**Review by dimas-b — APPROVED** [review]

File: polaris-core/src/main/java/org/apache/polaris/core/entity/PolarisMaintenanceProperties.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.entity;
+
+import static org.apache.polaris.core.entity.PolarisEntityConstants.MAINTENANCE_PREFIX;
+
+public enum PolarisMaintenanceProperties {
+  COMPACTION(MAINTENANCE_PREFIX + "compaction"),
+  SNAPSHOT_RETENTION(MAINTENANCE_PREFIX + "snapshot_expiration");
```
[the remark this answers is not in the document]
Comment (adutra) [reply polaris-core/src/main/java/org/apache/polaris/core/entity/PolarisMaintenanceProperties.java:25]: Also, the enum name says `RETENTION` but the property has `expiration` – would be good to standardize.

File: polaris-core/src/main/java/org/apache/polaris/core/entity/PolarisMaintenanceProperties.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+  private final String value;
```
Comment (adutra) [inline polaris-core/src/main/java/org/apache/polaris/core/entity/PolarisMaintenanceProperties.java:27]: the name `value` is misleading, it's rather the property _name_.

File: polaris-core/src/main/java/org/apache/polaris/core/entity/PolarisMaintenanceProperties.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+  PolarisMaintenanceProperties(String value) {
+    this.value = value;
+  }
+
+  public String getValue() {
+    return value;
+  }
+
+  public static PolarisMaintenanceProperties fromValue(String value) {
```
Comment (adutra) [inline polaris-core/src/main/java/org/apache/polaris/core/entity/PolarisMaintenanceProperties.java:37]: What is the added value of this method, as opposed to `PolarisMaintenanceProperties.valueOf()`?

**Comment by collado-mike** [comment]
Is this PR going to be closed due to the change in direction?

**Comment by flyrain** [comment]
Close it due to design change.

**Pull request closed without merging**